src/utils/slack/base.js

```diff
@@ -282,11 +288,84 @@ const wrapSayForThread = (say, threadTs) => {
   return wrappedFunction;
 };
 
+/**
+ * Downloads and parses a CSV file from Slack.
+ *
+ * @param {string} fileUrl - The private Slack file URL.
+ * @param {string} token - The Slack bot token for authentication.
+ * @returns {Promise<Array<Array<string>>>} Parsed CSV data as an array of rows.
+ * @throws {Error} - Throws an error if the file cannot be downloaded or parsed.
+ */
+const parseCSV = async (file, token) => {
+  try {
+    const fileUrl = file.url_private;
+    const response = await axios.get(fileUrl, {
+      headers: { Authorization: `Bearer ${token}` },
+      responseType: 'arraybuffer',
+      validateStatus: (status) => status < 500,
+    });
+
+    if (response.status === 401) throw new Error('Authentication failed: Invalid Slack token.');
+    if (response.status === 403) throw new Error('Access denied: Missing files:read permission.');
+    if (response.status === 404) throw new Error(`File not found at: ${fileUrl}.`);
+    if (!response.data || response.data.length === 0) {
+      throw new Error('CSV parsing resulted in empty or invalid data.');
+    }
+
+    const csvString = response.data.toString('utf-8').trim();
```

Comment:
i'd split this out into a separate function for separation of concerns. would make it more re-usable.